### PR TITLE
feat: Tool to repost PLE manually

### PR DESCRIPTION
**Pull request by ruthra-kumar**
## Repost PLE
Tool is primarily intended for manual correction of [Duplicate Payment Ledger entries on Item Value Repost](https://github.com/frappe/erpnext/pull/32567)

## Usage:
1. Use Filters to Bulk select or Add Vouchers manually and Submit 
Screenshot 2022-11-08 at 10 53 32 AM

2. Once submitted, a background job will be triggered. 
Screenshot 2022-11-08 at 10 53 41 AM

3. After successful run,`repost_status` will be updated 
Screenshot 2022-11-08 at 10 53 47 AM

`no-docs`

### CODE REVIEW

File: erpnext/accounts/doctype/repost_payment_ledger/repost_payment_ledger.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+
+class RepostPaymentLedger(Document):
+	def __init__(self, *args, **kwargs):
+		super(RepostPaymentLedger, self).__init__(*args, **kwargs)
+		self.vouchers = []
+
+	def validate(self):
+		if not self.company:
+			frappe.throw(_("Company is Mandatory"))
```
Comment (deepeshgarg007) [inline erpnext/accounts/doctype/repost_payment_ledger/repost_payment_ledger.py]: Needs checking again?
Both of these seem to have `reqd` as 1

File: erpnext/accounts/doctype/repost_payment_ledger/repost_payment_ledger.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+						docstatus: 1
+					}
+				}
+			}
+		}
+
+	},
+	refresh: function(frm) {
+
+		if (frm.doc.docstatus==1 && ['Queued', 'Failed'].find(x => x == frm.doc.repost_status)) {
```
Comment (deepeshgarg007) [inline erpnext/accounts/doctype/repost_payment_ledger/repost_payment_ledger.js:36]: This seems like an unnecessary click/step for the user I guess. The job should be by default Queued on submit and the status should be updated as "Queued". By default, the "Queued" status might be misleading to the user. Showing a progress bar will also be a good to have to show the user the realtime status of the job

Comment (ruthra-kumar) [reply erpnext/accounts/doctype/repost_payment_ledger/repost_payment_ledger.js:36]: I've changed it to trigger job on submit.

As repost happens in single transaction, Progress bar is not possible.

File: erpnext/accounts/doctype/repost_payment_ledger/repost_payment_ledger.py (hunk 1 above)
Comment (ruthra-kumar) [reply erpnext/accounts/doctype/repost_payment_ledger/repost_payment_ledger.py]: Removed

**Comment by deepeshgarg007** [comment]
@Mergifyio backport version-14-hotfix

**Comment by dawoodjee** [comment]
Which ERPNext versions has this been merged to?

I'm on v14.11 and its not working

First the function `is_job_queued` was absent in the latest stable Frappe 14.20. 

`if not frappe.utils.background_jobs.is_job_queued(job_name):`

After adding it, it just says its running in the background but not doing anything. I'll be debugging as I wait for a response

**Comment by ruthra-kumar** [comment]
> 

@dawoodjee This issue is fixed in v14.13.0

**Pull request merged**